File: frontend_beyond_detail/seo_data/analyze_new_pages.py
import json
import os

files = {
    "Ceramic Coating Toronto": "seo_data/lh_toronto.json",
    "Car Detailing Markham Road": "seo_data/lh_markham.json"
}

print(f"{'Page':<30} | {'Perf':<5} | {'SEO':<5} | {'Access':<6} | {'Best P':<6} | {'LCP':<10}")
print("-" * 85)

for name, filepath in files.items():
    if not os.path.exists(filepath):
        print(f"{name:<30} | FILE NOT FOUND")
        continue
        
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            data = json.load(f)
            
        cats = data.get('categories', {})
        perf = int(cats.get('performance', {}).get('score', 0) * 100)
        seo = int(cats.get('seo', {}).get('score', 0) * 100)
        acc = int(cats.get('accessibility', {}).get('score', 0) * 100)
        bp = int(cats.get('best-practices', {}).get('score', 0) * 100)
        
        lcp = data.get('audits', {}).get('largest-contentful-paint', {}).get('displayValue', 'N/A')
        
        print(f"{name:<30} | {perf:<5} | {seo:<5} | {acc:<6} | {bp:<6} | {lcp:<10}")
        
    except Exception as e:
        print(f"{name:<30} | Error: {str(e)}")

File: frontend_beyond_detail/seo_data/analyze_new_pages.js
const fs = require('fs');
const path = require('path');

const files = {
    "Ceramic Coating Toronto": "lh_toronto.json",
    "Car Detailing Markham Road": "lh_markham.json"
};

let output = `${"Page".padEnd(30)} | ${"Perf".padEnd(5)} | ${"SEO".padEnd(5)} | ${"Access".padEnd(6)} | ${"Best P".padEnd(6)} | ${"LCP".padEnd(10)}\n`;
output += "-".repeat(85) + "\n";

for (const [name, filename] of Object.entries(files)) {
    const fullPath = path.join(__dirname, filename); // Script is in seo_data, so just filename needed

    if (!fs.existsSync(fullPath)) {
        output += `${name.padEnd(30)} | FILE NOT FOUND: ${filename}\n`;
        continue;
    }

    try {
        const rawData = fs.readFileSync(fullPath);
        const data = JSON.parse(rawData);

        const cats = data.categories || {};
        const perf = Math.round((cats.performance?.score || 0) * 100);
        const seo = Math.round((cats.seo?.score || 0) * 100);
        const acc = Math.round((cats.accessibility?.score || 0) * 100);
        const bp = Math.round((cats['best-practices']?.score || 0) * 100);

        const lcp = data.audits?.['largest-contentful-paint']?.displayValue || 'N/A';

        output += `${name.padEnd(30)} | ${String(perf).padEnd(5)} | ${String(seo).padEnd(5)} | ${String(acc).padEnd(6)} | ${String(bp).padEnd(6)} | ${lcp.padEnd(10)}\n`;

    } catch (err) {
        output += `${name.padEnd(30)} | Error: ${err.message}\n`;
    }
}

fs.writeFileSync(path.join(__dirname, 'lh_summary_report.txt'), output);
console.log("Report generated.");
